src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -1599,17 +1599,18 @@ public static string FullTypeName(this IEdmType type)
                 return primitiveType.FullName;
             }
 

```

Comment:
enclose 1596-1600 in an if (type.TypeKind == EdmTypeKind.Primitive) to avoid the failed cast for non-primitive types